src/event/cmap.rs

```diff
@@ -144,6 +156,15 @@ pub struct ConnectionClosedEvent {
 
     /// The reason that the connection was closed.
     pub reason: ConnectionClosedReason,
+
+    /// If the `reason` connection checkout failed was `Error`,the associated
+    /// error is contained here. This is attached so we can include it in log messages;
+    /// in future work we may add this to public API on the event itself. TODO: add
```

Comment:
@patrickfreed was going to file a DRIVERS ticket about drivers adding these errors to their connection monitoring events, so once there is a ticket to link to I'll add that here. I figured for now we could just leave it out from the public API in case the eventual spec work leads to some different proposal to expose the error.  

ideally, we would probably actually add the error to the `ConnectionClosedReason::Error` enum case but that would be a breaking change.

Reply:
Ticket here: https://jira.mongodb.org/browse/DRIVERS-2495. Leaving it out until the RUST ticket for that sgtm

Reply:
added reference to DRIVERS-2495 here and below